File: src/test/com/intellij/lang/jsgraphql/editor/GraphQLQuoteHandlerTest.java
package com.intellij.lang.jsgraphql.editor;

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase;

public class GraphQLQuoteHandlerTest extends GraphQLTestCaseBase {

  public void testSingleQuote() {
    myFixture.configureByText("schema.graphql", "<caret> type Abc {}");
    myFixture.type('"');
    myFixture.checkResult("\"<caret>\" type Abc {}");
  }

  public void testSingleQuoteBackspace() {
    myFixture.configureByText("schema.graphql", "\"<caret>\" type Abc {}");
    myFixture.type('\b');
    myFixture.checkResult("<caret> type Abc {}");
  }
}

File: tests/testSrc/com/intellij/lang/jsgraphql/editor/GraphQLQuoteHandlerTest.kt
package com.intellij.lang.jsgraphql.editor

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase
import com.intellij.openapi.progress.runBlockingCancellable

class GraphQLQuoteHandlerTest : GraphQLTestCaseBase() {
  fun testSingleQuote() = runBlockingCancellable {
    myFixture.configureByText("schema.graphql", "<caret> type Abc {}")
    myFixture.type('"')
    myFixture.checkResult("\"<caret>\" type Abc {}")
  }

  fun testSingleQuoteBackspace() = runBlockingCancellable {
    myFixture.configureByText("schema.graphql", "\"<caret>\" type Abc {}")
    myFixture.type('\b')
    myFixture.checkResult("<caret> type Abc {}")
  }
}
